src/UnityExtension/Assets/Editor/GitHub.Unity/UI/Window.cs

```diff
@@ -227,16 +267,28 @@ private void AttachHandlers(IRepository repository)
         {
             if (repository == null)
                 return;
-            repository.OnRepositoryInfoChanged += RefreshOnMainThread;
-            repository.OnCurrentBranchUpdated += UpdateLog;
+            repository.CurrentBranchAndRemoteChanged += RepositoryOnCurrentBranchAndRemoteChanged;
+        }
+
+        private void RepositoryOnCurrentBranchAndRemoteChanged(CacheUpdateEvent cacheUpdateEvent)
+        {
+            if (!lastCurrentBranchAndRemoteChangedEvent.Equals(cacheUpdateEvent))
```

Comment:
Is this ever false? Isn't Repository supposed to only raise events if the data has changed (and therefore the cacheUpdateEvent is always different here?

Reply:
If you are looking at the `HistoryView` and switch to a new branch outside of Unity.
When you click on Unity, `Window` will update to the correct value.
Then switch to the `ChangesView` The changes view has an outdated value for `CurrentBranch`/`CurrentRemote`.
`ChangesView` will ask `Repository` to check it's cache timestamp.
`Repository` will see that the timestamp is out of date and re-fire the event.
in this scenario, line 275 will be false.

Reply:
Oooooh, yeah I see.